app/code/Magento/Catalog/Model/ProductRepository.php

```diff
@@ -856,4 +858,36 @@ private function saveProduct($product): void
             );
         }
     }
+
+    /**
+     * Join category position field to make sorting by position possible.
+     *
+     * @param Collection $collection
+     * @param SearchCriteriaInterface $searchCriteria
+     * @return void
+     */
+    private function joinPositionField(
+        Collection $collection,
+        SearchCriteriaInterface $searchCriteria
+    ): void {
+        $categoryIds = [[]];
+        foreach ($searchCriteria->getFilterGroups() as $filterGroup) {
```

Comment:
Warning: Invalid argument supplied for foreach() in /var/www/html/app/code/Magento/Catalog/Model/ProductRepository.php:874 on test https://public-results-storage-prod.magento-testing-service.engineering/reports/magento/magento2/pull/31629/1987de3af131eea1fc33bf4caa363d0b/Unit/allure-report-ce/index.html#categories/50ea141dbb5485cec1df0a24a3a2dc38/6184ee0e02b630f2/

you could wrap with:

if ($searchCriteria->getFilterGroups()) {
... foreach ...
}

Reply:
@danielrussob 
Thanks for review. There is no need to add such verification, $searchCriteria->getFilterGroups() returns array in any case. 
https://github.com/magento/magento2/blob/63434d2997f9625238d283d5dcaebfe918020546/lib/internal/Magento/Framework/Api/Search/FilterGroup.php#L29
Problem with unit test